src/main/scala/com/yotpo/metorikku/test/KeyColumns.scala

```diff
@@ -0,0 +1,63 @@
+package com.yotpo.metorikku.test
+import org.apache.spark.sql.DataFrame
+
+import scala.collection.mutable
+
+object KeyColumns {
+
+  def getKeyListFromMap(resultRows: List[Map[String, Any]], tableKeys: List[String]): Array[String] = {
+    resultRows.map(row => {
+      tableKeys.map(currKey => {
+        val res = if (row(currKey) == null) {
+          ""
+        } else {
+          row(currKey).toString
+        }
+        res
+      }).mkString("#")
+    }).toArray
+  }
+
+  def getKeyListFromDF(resultRows: DataFrame, tableKeys: List[String]): Array[String] = {
+    val metricActualResultsMap = TestUtil.getMapFromDf(resultRows)
+    KeyColumns.getKeyListFromMap(metricActualResultsMap, tableKeys)
+  }
+
+  def getRowKey(row: Map[String, Any], tableKeys: List[String]): String = {
+    tableKeys.map(key => {
+      row.getOrElse(key, 0).toString
+    }).mkString("#")
+  }
+
+  def formatRowOutputKey(row: Map[String, Any], tableKeys: List[String]): String = {
+    tableKeys.map { tableKey =>
+      row(tableKey) match {
+        case x if (x != null) => tableKey + "=" + x.toString
+        case _ => ""
+      }
+    }.mkString(", ")
+    //Key1=Value1, Key2=Value2
+  }
+
+  def formatOutputKey(key: String, tableKeys: List[String]): String = {
+    val outputVals = key.split("#")
```

Comment:
Not good... # can be an actual character in the data, you need to keep the array at all times... and just for the sorting purposes mkstring with the # for simplicity (you actually don't even have to do that, you can sort with the array)